Sources/LocalizedStringKit/LocalizedStringKit.m

```diff
@@ -12,38 +12,55 @@ @interface LocalizedStringKit : NSObject
 
 @implementation LocalizedStringKit
 
-NSString *Localized(NSString *value, NSString *comment) {
-  return [LocalizedStringKit localizeWithValue:value comment:comment keyExtension:nil];
+#pragma mark - Public
+
+NSString *Localized(NSString *_Nonnull value, NSString *_Nonnull comment, NSString *_Nullable tableName) {
+  return [LocalizedStringKit localizeWithValue:value comment:comment keyExtension:nil tableName:tableName];
 }
 
-NSString *LocalizedWithKeyExtension(NSString *value, NSString *comment, NSString *keyExtension) {
-  return [LocalizedStringKit localizeWithValue:value comment:comment keyExtension:keyExtension];
+NSString *LocalizedWithKeyExtension(NSString *_Nonnull value, NSString *_Nonnull comment, NSString *_Nonnull keyExtension, NSString *_Nullable tableName) {
+  return [LocalizedStringKit localizeWithValue:value comment:comment keyExtension:keyExtension tableName:tableName];
 }
 
 __attribute__((annotate("returns_localized_nsstring")))
 NSString *LocalizationUnnecessary(NSString *value) {
   return value;
 }
 
-NSBundle *getLocalizedStringKitBundle() {
-  return [LocalizedStringKit getLocalizedStringKitBundle];
+NSBundle *getLocalizedStringKitBundle(NSString *_Nullable tableName) {
+  return [LocalizedStringKit getLocalizedStringKitBundle:tableName];
 }
 
-+ (NSString *)localizeWithValue:(NSString *)value comment:(NSString *)comment keyExtension:(NSString *)keyExtension
+#pragma mark - Private / Static
+
++ (NSString *)localizeWithValue:(NSString *_Nonnull)value comment:(NSString *_Nonnull)comment keyExtension:(NSString *_Nullable)keyExtension tableName:(NSString *_Nullable)tableName
 {
   // Key
   NSString *key = [self keyWithValue:value keyExtension:keyExtension];
 
-  // Table
+  // Table: This does not change between bundles
   NSString *table = @"LocalizedStringKit";
 
-  // Bundle
-  static NSBundle *bundle = nil;
+  // Bundle Map: [tableName String: NSBundle]
+  static NSMutableDictionary<NSString *, NSBundle *> *bundleMap = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    bundleMap = [[NSMutableDictionary alloc] init];
+  });
+
+  if (tableName == nil)
+  {
+    // Default to primary strings bundle
+    tableName = @"LocalizedStringKit.bundle";
+  }
+
+  NSBundle *bundle = [bundleMap objectForKey:tableName];
```

Comment:
How thread-safe do we want this to be?

Reply:
The current degree is probably sufficient for the time being.